### PR TITLE
Additional queries and filters to support service/form config tools

### PR DESCRIPTION
**Pull request by gigxz**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Frontend PR is https://github.com/greenriver/hmis-frontend/pull/692, but this can be merged without it, as it does not have any breaking changes. The frontend PR is stacked on top of the Bed Night Pr. 

Functional updates to admin tool to allow more configuration to happen on the front-end:
* Support searching form definitions
* Support searching service types
* Support filtering on service types, hide HUD service types by default (since they will rarely be configured)
* Update Form Rule form JSON to support specifying Service Type or Category for form roles. This  is necessary to let us manage service form rules in the UI which we couldn't do before.
* Resolve relevant FormDefinitions on ServiceType so that the user can navigate over to the forms and manage rules there

These are functional changes to make configuration easier for us. They still need re-design before being used by an HMIS Admin.

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by gigxz** [comment]
This is low priority for review requests :)

File: drivers/hmis/app/graphql/types/hmis_schema/service_type.rb [hunk 1]
```diff
@@ -11,24 +11,44 @@ class HmisSchema::ServiceType < Types::BaseObject
...
 
       [object.hud_record_type, object.hud_type_provided].join(':')
     end
+
+    def form_definitions
+      raise 'unauthorized' unless current_user.can_configure_data_collection?
+
+      definitions_for_type = load_ar_association(object, :definitions)
+      definitions_for_category = load_ar_association(category_record, :definitions)
+      (definitions_for_type + definitions_for_category).uniq
```
Comment (ttoomey) [inline drivers/hmis/app/graphql/types/hmis_schema/service_type.rb]: should these be sorted?
```suggestion
      (definitions_for_type + definitions_for_category).uniq.sort_by(&:id)
```

File: drivers/hmis/app/graphql/types/hmis_schema/query_type.rb [hunk 2]
```diff
@@ -373,6 +373,17 @@ def service_categories
...
 
+    field :service_types, Types::HmisSchema::ServiceType.page_type, null: false do
+      filters_argument HmisSchema::ServiceType
+    end
+    def service_types(filters: nil)
+      raise 'Access denied' unless current_user.can_configure_data_collection?
+
+      scope = Hmis::Hud::CustomServiceType.all
+      scope = scope.apply_filters(filters) if filters
+      scope.order(:name)
```
Comment (ttoomey) [inline drivers/hmis/app/graphql/types/hmis_schema/query_type.rb]: I'm not sure if name is unique. Maybe include id as tie breaker:
```suggestion
      scope.order(:name, :id)
```

**Review by ttoomey — APPROVED** [review]
Looks good to me!

**Pull request merged**